src/main/java/core/basesyntax/strategy/operation/SupplyOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.Fruit;
+
+public class SupplyOperation implements OperationHandler {
+    private StorageDao storageDao = new StorageDaoImpl();
+
+    @Override
+    public void operationType(String fruit, int amount) {
+        if (storageDao.getFruit(fruit) != null) {
+            int newQuantity = storageDao.getFruit(fruit).getQuantity() + amount;
+            storageDao.update(fruit, newQuantity);
+        } else {
+            storageDao.addFruit(new Fruit(fruit, amount));
+        }
+    }
+}
```

Comment:
The `SupplyOperation` class is tightly coupled with `StorageDaoImpl`. It's better to inject the `StorageDao` through the constructor to follow the Dependency Inversion Principle of SOLID and make the class more testable.